pakyow-presenter/lib/presenter/binder.rb

```diff
@@ -59,16 +59,29 @@ def value_for_scoped_prop(scope, prop, bindable, bindings, context)
 
         if binding_fn
           binding_eval = BindingEval.new(prop, bindable, context)
-          binding_eval.instance_exec(binding_eval.value, bindable, context, &binding_fn)
-        else # default value
-          if bindable.is_a?(Hash)
-            bindable.fetch(prop) { bindable[prop.to_s] }
-          elsif bindable.class.method_defined?(prop)
-            bindable.send(prop)
+          result = binding_eval.eval(&binding_fn)
+
+          # TODO Decide if bindings given as arguments should also only use
```

Comment:
I wouldn't add warnings unless we were planning to deprecate hash support. And I don't think we can deprecate that completely because it's possible to pass bindings into the view transformation methods (e.g. `bind`, `apply`).

One question this raises -- can we still respect the `data-parts-*` attributes when not defining bindings using parts? I want the behavior to be as consistent as possible. Ideally we can respect the `data-parts-*` attributes in all cases, then push people to use the `part` syntax when creating bindings in the normal way.